lib/middlewares/src/metering.rs

```diff
@@ -173,14 +183,20 @@ impl<F: Fn(&Operator) -> u64 + Copy + Clone + Send + Sync> FunctionMiddleware
 ///
 /// The instance Module must have been processed with the [`Metering`] middleware
 /// at compile time, otherwise this will panic.
-pub fn get_remaining_points(instance: &Instance) -> u64 {
-    instance
+pub fn get_remaining_points(instance: &Instance) -> MeteringPoints {
+    let points: i64 = instance
         .exports
-        .get_global("remaining_points")
-        .expect("Can't get `remaining_points` from Instance")
+        .get_global("wasmer_metering_remaining_points")
+        .expect("Can't get `wasmer_metering_remaining_points` from Instance")
         .get()
         .try_into()
-        .expect("`remaining_points` from Instance has wrong type")
+        .expect("`wasmer_metering_remaining_points` from Instance has wrong type");
+
+    if points < 0 {
```

Comment:
With this single i64/u64 value, any u64 value geater than i64::MAX is interpreted as a negative value and error. E.g.

```rust
set_remaining_points(&instance, u64::MAX);
get_remaining_points(&instance) // depletion
```

I'd rather stick to two variables than limiting the values the user can use